Web Inspector: Use-after-scope in lambda in WebCore::isValidRuleHeaderText()

ddkilzer opened the pull request:
#### 599c0cb20b8118a147f7ef235e56d3dcb2fe7758
<pre>
Web Inspector: Use-after-scope in lambda in WebCore::isValidRuleHeaderText()
<a href="https://bugs.webkit.org/show_bug.cgi?id=249119">https://bugs.webkit.org/show_bug.cgi?id=249119</a>
&lt;rdar://102887283&gt;

Reviewed by Patrick Angle.

`CSSParserImpl` holds a reference to a `CSSParserContext`, but
isValidRuleHeaderText() creates a temporary context on the
stack, which is destroyed after the parser is created.  Fix this
by creating a stack variable for the context that outlives the
parser.

Test: inspector/css/setGroupingHeaderText.html

* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::isValidRuleHeaderText):

Canonical link: <a href="https://commits.webkit.org/257722@main">https://commits.webkit.org/257722@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/df7c911f5022485f0e93471772f56b1bbf976e5d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99791 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8971 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32879 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109157 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169396 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/103794 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/9728 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86262 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92255 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107067 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105558 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90713 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34173 "Passed tests") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22097 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2787 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23609 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2733 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45988 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5303 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/8868 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43078 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4596 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

patrickangle reviewed (approved):
Oops - thank you for catching this!

Commit:
Web Inspector: Use-after-scope in lambda in WebCore::isValidRuleHeaderText()
https://bugs.webkit.org/show_bug.cgi?id=249119
<rdar://102887283>

Reviewed by Patrick Angle.

`CSSParserImpl` holds a reference to a `CSSParserContext`, but
isValidRuleHeaderText() creates a temporary context on the
stack, which is destroyed after the parser is created.  Fix this
by creating a stack variable for the context that outlives the
parser.

Test: inspector/css/setGroupingHeaderText.html

* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::isValidRuleHeaderText):

Canonical link: https://commits.webkit.org/257722@main
